Automattic/jetpack
# JS Components: Add Dialog component

This is the last PR that introduces the `<Dialog />` component to the jetpack-components packages. More detail about implementation https://github.com/Automattic/jetpack/issues/23754.

Fixes https://github.com/Automattic/jetpack/issues/23754

#### Changes proposed in this Pull Request:

* JS Components: Add Dialog component

#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?



#### Testing instructions:






* Use Storybook
* Go to Dialog story
* Take a look at both stories
* Confirm the box-model is consistent with the design

<img width="1087" alt="image" src="https://user-images.githubusercontent.com/77539/162265318-0b110244-17b6-40ba-96c2-188ede25dcee.png">

<img width="1012" alt="Screen Shot 2022-04-07 at 2 49 57 PM" src="https://user-images.githubusercontent.com/77539/162265635-72da614a-35e0-46e6-9414-101f5f180c03.png">


https://user-images.githubusercontent.com/77539/162266031-96136676-2aff-4b05-b30b-16e5447a2c54.mov
